lib/cloud/aws/policy_statements.go

```diff
@@ -0,0 +1,100 @@
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+var (
+	allResources = []string{"*"}
+)
+
+// StatementForIAMEditRolePolicy returns a IAM Policy Statement which allows editting Role Policy
+// of the resources.
+func StatementForIAMEditRolePolicy(resources ...string) *Statement {
+	return &Statement{
+		Effect:    EffectAllow,
+		Actions:   []string{"iam:GetRolePolicy", "iam:PutRolePolicy", "iam:DeleteRolePolicy"},
+		Resources: resources,
+	}
+}
+
+// StatementForIAMEditUserPolicy returns a IAM Policy Statement which allows editting User Policy
+// of the resources.
+func StatementForIAMEditUserPolicy(resources ...string) *Statement {
+	return &Statement{
+		Effect:    EffectAllow,
+		Actions:   []string{"iam:GetUserPolicy", "iam:PutUserPolicy", "iam:DeleteUserPolicy"},
+		Resources: resources,
+	}
+}
+
+// StatementForECSDeployServicePolicy returns the statement that allows managing the ECS Service deployed
+// by DeployService (AWS OIDC Integration).
+func StatementForECSManageService() *Statement {
+	return &Statement{
+		Effect: EffectAllow,
+		Actions: []string{
+			"ecs:DescribeClusters", "ecs:CreateCluster", "ecs:PutClusterCapacityProviders",
+			"ecs:DescribeServices", "ecs:CreateService", "ecs:UpdateService",
+			"ecs:RegisterTaskDefinition",
+		},
+		Resources: allResources,
+	}
+}
+
+// StatementForLogsDeployServicePolicy returns the statement that allows the writing logs to CloudWatch.
+// This is used by the DeployService (ECS Service) to write teleport logs.
+// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html
+func StatementForWritingLogs() *Statement {
+	return &Statement{
+		Effect:    EffectAllow,
+		Actions:   []string{"logs:CreateLogStream", "logs:PutLogEvents", "logs:CreateLogGroup"},
+		Resources: allResources,
+	}
+}
+
+// StatementForIAMPassRole returns a statement that allows to iam:PassRole the target role.
+// Usage example: when setting up the TaskRole for the ECS Task.
+// https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-iam-roles.html#specify-task-iam-roles
+func StatementForIAMPassRole(targetRole string) *Statement {
+	return &Statement{
+		Effect:  EffectAllow,
+		Actions: SliceOrString{"iam:PassRole"},
+		Resources: SliceOrString{
+			targetRole,
+		},
+	}
+}
+
+// StatementForECSTasksAssumeRole returns the Trust Relationship to allow the ECS Tasks service to.
+// It allows the usage of this Role by the ECS Tasks service.
+func StatementForECSTasksAssumeRole() *Statement {
```

Comment:
nit: `StatementForECSTasksAssumeRole` may sound like the task role will assume another role. Maybe call it something like `StatementForECSTaskRoleTrustRelationships`